scanner/dockerfile.go

```diff
@@ -7,34 +7,45 @@ import (
 	"strconv"
 )
 
-func configureDockerfile(sourceDir string, config *ScannerConfig) (*SourceInfo, error) {
+const defaultPort = 8080
+
+func configureDockerfile (sourceDir string, config *ScannerConfig) (*SourceInfo, error) {
 	if !checksPass(sourceDir, fileExists("Dockerfile")) {
 		return nil, nil
 	}
 
+	var portFromDockerfile int
+
 	s := &SourceInfo{
 		DockerfilePath: filepath.Join(sourceDir, "Dockerfile"),
 		Family:         "Dockerfile",
+		Port : config.ExistingPort,
 	}
 
-	// extract port number from EXPOSE statement
-	dockerfile, err := os.ReadFile("Dockerfile")
-	if err == nil {
-		port := 8080
-		re := regexp.MustCompile(`(?m)^EXPOSE\s+(?P<port>\d+)`)
-		m := re.FindStringSubmatch(string(dockerfile))
-
-		for i, name := range re.SubexpNames() {
-			if len(m) > 0 && name == "port" {
-				port, err = strconv.Atoi(m[i])
-				if err != nil {
-					panic(err)
-				} else if port == 8080 {
-					break
-				}
+	dockerfile, err := os.ReadFile(s.DockerfilePath)
+	if err != nil {
+		// just maintaining existing behaviour from old code.
+		return s, nil
+	}
+
+	re := regexp.MustCompile(`(?m)^EXPOSE\s+(?P<port>\d+)`)
```

Comment:
it is better placed as a package constant at the top of the file. It will fail to compile if the regex is invalid

Reply:
i'm not sure i follow. I can't set it to a constant either because that variable("re"), is actually mutated later on by the regexp pacakage.  I've moved it to the top of the file though.

Reply:
My bad, you are right. I said "constant" where the intention was to declare it as a package private variable at the top of the file. 👍